pokt-foundation/explorer
# [Refactor]: Call Gateway for QoS Controls

Use gateway middleware as data source instead of direct pocket network through pocket-js.

Replacing the following methods:

- [x] getHeight
- [x] getAccount
- [x] getTransaction

- [x] getBlock

- [x] getLatestBlock

- [x] getLatestTransactions

- [x] getTotalStakedApps

- [x] getStakedSupply

- [x] getNodes


## How To Test Locally

1- Update your src/config/config.json to use the production gateway url with an app id staked for 0001
2- Run `npm install` to install the newly added dependencies
3- Run `npm run test ./src/datasource/gateway.test.js` to see whether the gateway client is functional with all endpoints
4- Run `npm run test ./src/datasource/datasource.test.js` to see whether the datasource behaviors properly while using the gateway client
5- Run `npm run start` to test in action
